### PR TITLE
Add tests for incremental reuse scenarios

### PR DESCRIPTION
**Pull request by nikomatsakis**
These are microbenchmarks checking that we achieve the expected reuse in the scenarios covered by incremental beta.

r? @michaelwoerister

### CODE REVIEW

File: src/test/incremental/change_add_field/struct_point.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+mod fn_with_type_in_body {
+    use point::Point;
+
+    #[rustc_dirty(label="TypeckItemBody", cfg="rpass2")]
+    pub fn boop() -> f32 {
+        Point::origin().total()
+    }
+}
+
+/// A fn item that calls (public) methods on `Point` from the same impl which changed
```
Comment (michaelwoerister) [inline src/test/incremental/change_add_field/struct_point.rs]: This comment seems wrong.

File: src/test/incremental/change_pub_inherent_method_body/struct_point.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Test where we change the body of a private method in an impl.
```
Comment (michaelwoerister) [inline src/test/incremental/change_pub_inherent_method_body/struct_point.rs]: But the changed method is public?

File: src/test/incremental/change_pub_inherent_method_sig/struct_point.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Test where we change the body of a private method in an impl.
+// We then test what sort of functions must be rebuilt as a result.
```
Comment (michaelwoerister) [inline src/test/incremental/change_pub_inherent_method_sig/struct_point.rs]: Outdated comment?

**Review by michaelwoerister — COMMENTED** [review]
r=me with the comments fixed.

**Comment by nikomatsakis** [comment]
@michaelwoerister oh yeah shoot I forgot to go fixup the comments

**Pull request merged**